### PR TITLE
Remove dockstore-webservice dependency

### PR DESCRIPTION
**Pull request by kathy-t**
**Description**
Companion webservice PR https://github.com/dockstore/dockstore/pull/5562

This PR removes the dockstore-webservice dependency by only using things from dockstore-common. Note that the metrics-aggegator still uses the dockstore-webservice with the test scope.

I also removed all javax dependencies because I didn't see any imports for them. Looks like javax imports were deleted in https://github.com/dockstore/dockstore-support/pull/468 when we removed the jenkins functionality. This will remove the need for https://github.com/dockstore/dockstore/issues/5527.

**Review Instructions**
Tests should pass and the metricsaggregator and tooltester should still be functional. Can try running a command for both the metrics aggregator and tooltester to make sure the JAR works. For the metrics aggregator, could submit validation data using a simple CSV file containing data for one workflow. For the tooltester, could upload metrics.

**Issue**
dockstore/dockstore#5528
https://ucsc-cgl.atlassian.net/browse/DOCK-2404

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install` in the project that you have modified (until https://ucsc-cgl.atlassian.net/browse/SEAB-5300 adds multi-module support properly)
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If you are changing dependencies, check with dependabot to ensure you are not introducing new high/critical vulnerabilities
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

### CODE REVIEW

File: THIRD-PARTY-LICENSES.txt [hunk 1]
```diff
@@ -1,14 +1,14 @@
 
-Lists of 411 third-party dependencies.
+Lists of 399 third-party dependencies.
```
Comment (denis-yuen) [inline THIRD-PARTY-LICENSES.txt:2]: 12 down, I see things like `dockstore-integration-testing` and `dropwizard-testing` which makes me think a lot of what I thought would be removed is staying around from a licenses POV due to that test scope. 

What is the difference in jar size (or maven shade plugin warning)?

Comment (denis-yuen) [reply THIRD-PARTY-LICENSES.txt:2]: Also relevant to https://github.com/dockstore/dockstore/issues/5549

File: pom.xml [hunk 2]
```diff
@@ -598,9 +541,9 @@
                                         <!-- superceded by org.apache.httpcomponents:httpclient -->
                                         <exclude>commons-httpclient:commons-httpclient</exclude>
                                         <!-- see https://github.com/dropwizard/dropwizard/pull/3069 -->
-                                        <!-- used by toolbackup <exclude>javax.annotation:javax.annotation-api</exclude> -->
-                                        <!-- still used by tooltester <exclude>javax.activation:activation</exclude> -->
-                                        <!-- still used by tooltester <exclude>javax.ws.rs:javax.ws.rs-api</exclude> -->
+                                        <exclude>javax.annotation:javax.annotation-api</exclude>
```
Comment (denis-yuen) [inline pom.xml:544]: Nice

File: toolbackup/pom.xml [hunk 3]
```diff
@@ -170,12 +170,14 @@
...
             <groupId>com.amazonaws</groupId>
             <artifactId>aws-java-sdk-s3</artifactId>
+            <version>1.11.83</version>
         </dependency>
 
         <!-- https://mvnrepository.com/artifact/com.amazonaws/aws-java-sdk-core -->
         <dependency>
             <groupId>com.amazonaws</groupId>
             <artifactId>aws-java-sdk-core</artifactId>
+            <version>1.11.83</version>
```
Comment (denis-yuen) [inline toolbackup/pom.xml:180]: Odd, these two should be in our bom via the AWS bom 
https://github.com/dockstore/dockstore/blob/develop/bom-internal/pom.xml#L337-L342

**Review by denis-yuen — COMMENTED** [review]
Likely ok, but will return after reviewing webservice PR

File: toolbackup/pom.xml (hunk 3 above)
Comment (kathy-t) [reply toolbackup/pom.xml:180]: Looking at the [package names](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/migration-whats-different.html), our bom has the AWS version 2 SDK, but toolbackup uses version 1

File: THIRD-PARTY-LICENSES.txt (hunk 1 above)
Comment (denis-yuen) [reply THIRD-PARTY-LICENSES.txt:2]: A develop webservice is 159M, I think that drop is significant.

**Review by denis-yuen — APPROVED** [review]
After the other one of course

**Comment by kathy-t** [comment]
I found a problem when running the tooltester and metrics aggregator so I removed reviewers and made this PR a draft. Will re-request reviews when it's ready again

**Comment by kathy-t** [comment]
When running the metrics aggregator and tool tester locally, they failed. Turns out they need the `org.glassfish.jersey.inject:jersey-hk2` dependency to run. Disclaimer: I only tested the metrics aggregator and tooltester commands, not the toolbackup, but it seems like we currently don't use it.

I ran into a null pointer exception when running workflows on AGC using the tooltester so I added a null check.

File: metricsaggregator/pom.xml [hunk 4]
```diff
@@ -89,35 +89,20 @@
...
-                <exclusion>
-                    <groupId>com.fasterxml.jackson.core</groupId>
-                    <artifactId>jackson-databind</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>com.fasterxml.jackson.module</groupId>
-                    <artifactId>jackson-module-jaxb-annotations</artifactId>
-                </exclusion>
-            </exclusions>
+            <scope>test</scope>
```
Comment (denis-yuen) [inline metricsaggregator/pom.xml:105]: :partying_face:

File: toolbackup/pom.xml [hunk 5]
```diff
@@ -170,12 +170,14 @@
         <dependency>
             <groupId>com.amazonaws</groupId>
             <artifactId>aws-java-sdk-s3</artifactId>
+            <version>1.11.83</version>
```
Comment (denis-yuen) [inline toolbackup/pom.xml:173]: Can use a property for these two to hint to a future developer that either these two are upgraded at once or more likely, deleted at the same time

File: tooltester/pom.xml [hunk 6]
```diff
@@ -169,37 +166,11 @@
                 </exclusion>
             </exclusions>
         </dependency>
-        <dependency>
```
Comment (denis-yuen) [inline tooltester/pom.xml:172]: :+1:

**Review by denis-yuen — APPROVED** [review]
Thanks for checking for runtime errors, those can be pretty annoying to track down since they don't show up in tests

**Review by coverbeck — APPROVED** [review]

**Review by svonworl — APPROVED** [review]

**Comment by kathy-t** [comment]
Updated the dockstore-core dependency to a tag. Build failed because of the new cwlavro version in the webservice, so I updated the pom to use the cwlavro dependencies from the bom

**Review by denis-yuen — APPROVED** [review]

**Pull request merged**